src/kademlia/item.cpp

```diff
@@ -140,10 +140,12 @@ item::item(entry v)
 {}
 
 item::item(bdecode_node const& v)
-	: m_value(std::move(v))
-	, m_seq(0)
+	: m_seq(0)
 	, m_mutable(false)
-{}
+{
+	// TODO: implement ctor for entry from bdecode_node?
+	m_value = v;
```

Comment:
m_value is an `entry`. I assume the move fell back to a copy then used the bool conversion on `bdecode_node` to construct an `entry` from integer.